prefect_docker/worker.py

```diff
@@ -423,6 +413,58 @@ async def run(
             identifier=container_pid,
         )
 
+    async def kill_infrastructure(
+        self,
+        infrastructure_pid: str,
+        configuration: DockerWorkerJobConfiguration,
+        grace_seconds: int = 30,
+    ):
+        """
+        Stops a container for a cancelled flow run based on the provided infrastructure
+        PID.
+        """
+        docker_client = self._get_client()
+
+        base_url, container_id = self._parse_infrastructure_pid(infrastructure_pid)
+        if docker_client.api.base_url != base_url:
+            raise InfrastructureNotAvailable(
+                "".join(
+                    [
+                        (
+                            f"Unable to stop container {container_id!r}: the current"
+                            " Docker API "
+                        ),
+                        (
+                            f"URL {docker_client.api.base_url!r} does not match the"
+                            " expected "
+                        ),
+                        f"API base URL {base_url}.",
+                    ]
+                )
+            )
+        await run_sync_in_worker_thread(
+            self._stop_container, container_id, docker_client, grace_seconds
+        )
+
+    def _stop_container(
+        self,
+        container_id: str,
+        client: "DockerClient",
+        grace_seconds: int = 30,
+    ):
+        try:
+            container = client.containers.get(container_id=container_id)
+        except docker.errors.NotFound:
+            raise InfrastructureNotFound(
+                f"Unable to stop container {container_id!r}: The container was not"
+                " found."
+            )
+
+        try:
+            container.stop(timeout=grace_seconds)
```

Comment:
What's the reason for try/excepting here if it's just going to raise anyways?

Reply:
That's silly. I must've left that in after some refactoring.